sdks/python/apache_beam/transforms/core.py

```diff
@@ -585,7 +585,7 @@ def default_type_hints(self):
       try:
         fn_type_hints.strip_iterable()
       except ValueError as e:
-        raise ValueError('Return value not iterable: %s: %s' % (self, e))
+        logging.warning('%s: %s', self.default_label(), e)
```

Comment:
It's unclear to me under what conditions it's OK for this not to be iterable.

Reply:
I automatically made this a warning to be consistent with the warning for the other invocation of strip_iterable below.
Below, the code historically allowed output type hints of DoFns to not be iterable. I would like it to be an exception when the output type is not iterable, but it seems that it would break existing pipelines (at least in the case below).

Reply:
I've reverted this case back to raise an exception.